integration/integration_memberlist_single_binary_test.go

```diff
@@ -25,10 +40,37 @@ func TestSingleBinaryWithMemberlist(t *testing.T) {
 	require.NoError(t, s.StartAndWaitReady(dynamo))
 
 	require.NoError(t, writeFileToSharedDir(s, cortexSchemaConfigFile, []byte(cortexSchemaConfigYaml)))
-
-	cortex1 := newSingleBinary("cortex-1", "")
-	cortex2 := newSingleBinary("cortex-2", networkName+"-cortex-1:8000")
-	cortex3 := newSingleBinary("cortex-3", networkName+"-cortex-1:8000")
+	var cortex1, cortex2, cortex3 *e2ecortex.CortexService
+	if tlsEnabled {
+		// set the ca
+		cert := ca.New("single-binary-memberlist")
+
+		// Ensure the entire path of directories exist.
+		require.NoError(t, os.MkdirAll(filepath.Join(s.SharedDir(), "certs"), os.ModePerm))
+
+		require.NoError(t, cert.WriteCACertificate(filepath.Join(s.SharedDir(), caCertFile)))
+		require.NoError(t, cert.WriteCertificate(
+			&x509.Certificate{
+				Subject: pkix.Name{CommonName: "memberlist"},
+				DNSNames: []string{
+					networkName + "-cortex-1",
+					networkName + "-cortex-2",
+					networkName + "-cortex-3",
+				},
+				ExtKeyUsage: []x509.ExtKeyUsage{x509.ExtKeyUsageAny},
+			},
+			filepath.Join(s.SharedDir(), clientCertFile),
+			filepath.Join(s.SharedDir(), clientKeyFile),
+		))
+
+		cortex1 = newSingleBinary("cortex-1", networkName+"-cortex-1", "")
+		cortex2 = newSingleBinary("cortex-2", networkName+"-cortex-2", networkName+"-cortex-1:8000")
+		cortex3 = newSingleBinary("cortex-3", networkName+"-cortex-3", networkName+"-cortex-1:8000")
```

Comment:
This maybe a bit misleading. The TLSServerName should be set to a name required on the certificate of the peer (you are setting the name to ourselves).

So in the case of `cortex1` (line 66), I would expect it to be set to cortex-2 and cortex-3. 

Probably the best is only have a single DNS name on the certificate as otherwise we would need to set it independently depending on which peer we are connecting to.